add reload metadata command to the CLI

abilash13 opened the pull request:
<!-- Thank you for submitting this PR! :) -->
<!-- Provide a general summary of your changes in the Title above, end with (close #<issue-no>) or (fix #<issue-no>) -->

### Description
<!-- Describe your changes in detail -->
Added reload metadata command to the CLI

<!-- Please put an `x` in the boxes below -->

What component does this PR affect? 

- [ ] Server
- [ ] Console
- [x] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

### Related Issue
<!-- Please make sure you have an issue associated with this Pull Request -->
<!-- If you are suggesting a new feature or change, please discuss it in an issue first -->
<!-- If you are fixing a bug, there should be an issue describing it with steps to reproduce -->
<!-- Please don't forget to add `(close/fix #<issue-no>)` to the pull request title -->

<!-- Please link to the issue here: -->
#722 

### Solution and Design
<!-- How is this issue solved/fixed? What is the design? -->
<!-- It's better if we elaborate -->

1. Added a new method called ReloadMetadata to MetadataDriver.
2. Implemented the ReloadMetadata method (similar to ResetMetadata) with the query:
```
query := HasuraInterfaceQuery{
Type: "reload_metadata",
Args: HasuraArgs{},
}
```
3. Created a metadata_reload (similar to metadata_reset).
4. Created a metadata_reload_test as well.

(Thanks to @shahidhk for showing the steps)

### Type
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] Community content

### Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [x] This change requires a change in the documentation. 
- [ ] I have updated the documentation accordingly.
- [x] I have added required tests.

shahidhk commented:
@abilash13 I forgot to mention that you have to register the command in `metadata.go`. Will you be able to do that?

abilash13 commented:
I have added the command inside `metadataCmd.AddCommand` in `cli/commands/metadata.go`

Just a quick question: I have also added this inside `cli/commands/migrate.go`:

`func (m *Migrate) ReloadMetadata() error {
	return m.databaseDrv.ReloadMetadata()
}`

And then added a case for reload inside executeMetadata inside `cli/commands/migrate.go`?

`m.databaseDrv.ReloadMetadata()` should work right?

(Sorry if I'm asking such basic questions)
